src/cls/rgw/cls_rgw.cc

```diff
@@ -3219,7 +3219,8 @@ static int gc_iterate_entries(cls_method_context_t hctx, const string& marker, b
       if (max_entries && (i >= max_entries)) {
         if (truncated)
           *truncated = true;
-        key_iter = key;
+        --iter;
+        key_iter = iter->first;
```

Comment:
can you explain the `--iter` here? doesn't `key` represent the `next_marker` that we want to use when resuming the iteration?